src/pages/task/metadata/Stepback.tsx

```diff
@@ -0,0 +1,70 @@
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { InfoSprinkle } from "@leafygreen-ui/info-sprinkle";
+import { Skeleton } from "@leafygreen-ui/skeleton-loader";
+import { BaseFontSize } from "@leafygreen-ui/tokens";
+import { MetadataItem } from "components/MetadataCard";
+import { size } from "constants/tokens";
+import { TaskQuery } from "gql/generated/types";
+import { useBreakingTask } from "hooks/useBreakingTask";
+import { useLastPassingTask } from "hooks/useLastPassingTask";
+
+type Props = {
+  taskId: string;
+};
+
+/**
+ * Returns whether the task is in stepback.
+ * @param task The task to check if it is in stepback.
+ * @returns Whether the task is in stepback.
+ */
+export function isInStepback(task: TaskQuery["task"]) {
+  // The 'lastFailingStepbackTaskId' is set for all stepback tasks except the first one.
+  const hasLastStepback =
+    task?.stepbackInfo?.lastFailingStepbackTaskId?.length > 0;
+
+  // The 'nextStepbackTaskId' is only set when the next task in stepback is running/finished.
+  // This happens in the beginning of stepback or them middle of stepback. This condition is
+  // covering for the beginning of stepback.
+  const isBeginningStepback =
+    task?.stepbackInfo?.nextStepbackTaskId !== undefined &&
+    task?.stepbackInfo?.nextStepbackTaskId !== "";
```

Comment:
Can this be rewritten with length as well, or did you run into a problem? 👀

Reply:
Nope! Just got confused but I updated it to use the length trick too